Assets/HoloToolkit/Input/Scripts/Focus/InputSourcePointer.cs

```diff
@@ -20,41 +21,61 @@ public class InputSourcePointer : IPointingSource
 
         public bool OwnAllInput { get; set; }
 
-        public Ray Ray
-        {
+        [Obsolete("Will be removed in a later version. Use Rays instead.")]
+        public Ray Ray { get { return Rays[0]; } }
+
+        public RayStep[] Rays {
             get
             {
-                return (RayStabilizer == null)
-                    ? rawRay
-                    : RayStabilizer.StableRay;
+                return rays;
             }
         }
 
+        public PointerResult Result { get; set; }
+
         public float? ExtentOverride { get; set; }
 
         public LayerMask[] PrioritizedLayerMasksOverride { get; set; }
 
-        private Ray rawRay = default(Ray);
+        public bool InteractionEnabled
+        {
+            get
+            {
+                return true;
+            }
+        }
 
-        public void UpdatePointer()
+        private RayStep[] rays = new RayStep[1] { new RayStep(Vector3.zero, Vector3.forward) };
+        
+        public void OnPreRaycast()
         {
             if (InputSource == null)
             {
-                rawRay = default(Ray);
+                rays[0] = default(RayStep);
             }
             else
             {
                 Debug.Assert(InputSource.SupportsInputInfo(InputSourceId, SupportedInputInfo.Pointing));
 
-                InputSource.TryGetPointingRay(InputSourceId, out rawRay);
+                Ray pointingRay = default(Ray);
+                if (InputSource.TryGetPointingRay(InputSourceId, out pointingRay))
+                {
+                    rays[0].CopyRay(pointingRay, FocusManager.Instance.GetPointingExtent (this));
+                }
             }
 
             if (RayStabilizer != null)
             {
-                RayStabilizer.UpdateStability(rawRay.origin, rawRay.direction);
+                RayStabilizer.UpdateStability(rays[0].origin, rays[0].direction);
+                rays[0].CopyRay(RayStabilizer.StableRay, FocusManager.Instance.GetPointingExtent(this));
             }
         }
 
+        public void OnPostRaycast()
+        {
+            // Nothing needed
```

Comment:
Should we make this virtual in case someone wants to do something here?
Should we also make the `PrePostRaycast` virtual as well?

Reply:
Sure. I had assumed InputSourcePointer and GazeManager were more one-off classes than something folks would try to inherit from. But if you think that's a valid use case they should be virtual.

Reply:
Some people like to inherit and overwrite their own logic for some classes.  See #1265